SSA/Experimental/Bits/Fast/Decide.lean

```diff
@@ -23,154 +13,41 @@ instance (p : Predicate) :
-def z := Term.var 2
-
-example : ((and x y) + (or x y)).eval = (x + y).eval := by
-  native_decide
-
-example : ((or x y) - (xor x y)).eval = (and x y).eval := by
-  native_decide
-
-
-/-
-Note that we use `#eval!` instead of `#eval`, since our
-proof of `decide` currently contains a `sorry`. We should elminate the `sorry`,
-and then switch to `#eval`.
--/
-
-/--
-info: 'Decidable.decide' does not depend on any axioms
----
-info: 'decide' depends on axioms: [propext, sorryAx, Classical.choice, Quot.sound]
--/
-#guard_msgs in #print axioms decide
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + -x) 0
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (incr x) (x + 1)
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (decr x) (x - 1)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + -y) (x - y)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + 0) (var 0)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + y) (y + x)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + (y + z)) (x + y + z)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x - x) 0
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + 0) x
-
-/-- info: true -/
-#guard_msgs in #eval! decide (0 + x) x
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (-x) (not x).incr
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (-x) (not x.decr)
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (not x) (-x).decr
-
-/-- info: true -/
-#guard_msgs in #eval! decide (-not x) (x + 1)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + y) (x - not y - 1)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + y) ((xor x y) + (and x y).ls false)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + y) (or x y + and x y)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x + y) ((or x y).ls false - (xor x y))
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (x - y) (x + not y).incr
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x - y) (xor x y - (and (not x) y).ls false)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x - y) (and x (not y) - (and (not x) y))
-
-/-- info: true -/
-#guard_msgs in #eval! decide (x - y) ((and x (not y)).ls false - (xor x y))
-
-/-- info: true -/
-#guard_msgs in #eval! decide (xor x y) ((or x y) - (and x y))
-
-/-- info: true -/
-#guard_msgs in #eval! decide (and x (not y)) (or x y - y)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (and x (not y)) (x - and x y)
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (not (x - y)) (y - x).decr
-
-/-- info: true -/
-#guard_msgs in #eval! decide (not (x - y)) (not x + y)
-
--- /-- info: true -/
--- #guard_msgs in #eval! decide (not (xor x y)) (and x y - (or x y)).decr
-
-/-- info: true -/
-#guard_msgs in #eval! decide (not (xor x y)) (and x y + not (or x y))
-
-/-- info: true -/
-#guard_msgs in #eval! decide (or x y) (and x (not y) + y)
-
-/-- info: true -/
-#guard_msgs in #eval! decide (and x y) (or (not x) y - not x)
```

Comment:
Why do you drop the test cases?

Reply:
Because they have been ported to `Bits/Fast/Reflect.lean`.

This tests the old decision procedure, which tries to verify `t1.eval = t2.eval`. I drop support for this to get an end-to-end, sorry free proof.

Note that our current framework subsumes this, as we can simulate `t1.eval = t2.eval` with `Predicate.eval (Predicate.eq t1 t2)`.